docs/pages/guides/environment-variables.mdx

```diff
@@ -52,67 +36,89 @@ function Post() {
   return <Button onPress={onPress} title="Post" />;
 }
 ```
+When you run `npx expo start`, `process.env.EXPO_PUBLIC_API_URL` will be replaced with `https://staging.example.com` in your app bundle. Variables can be updated as you edit your code without restarting the Expo CLI or clearing the cache. You will need to perform a full reload (e.g., shake gesture and then Reload in Expo Go or your development build) to see the updated value.
+
+> **warning** Do not store sensitive info, such as private keys, in `EXPO_PUBLIC_` variables. These variables will be visible in plain-text in your compiled application.
+
+### How variables are loaded
+Metro Bundler loads **.env** files according to the [standard .env file resolution](https://github.com/bkeepers/dotenv/blob/c6e583a/README.md#what-other-env-files-can-i-use) and then replaces all references in your code to `process.env.EXPO_PUBLIC_[VARNAME]` with the corresponding value set in the **.env** files. Code inside **node_modules** is not affected.
+
+Every variable must be referenced as a static property in order for it to be inlined. For example, the expression `process.env.EXPO_PUBLIC_KEY` will be rewritten, but `process.env['EXPO_PUBLIC_KEY']` will not.
+
+### Using multiple .env files to define separate environments
+You can define any of the [standard .env files](https://github.com/bkeepers/dotenv/blob/c6e583a/README.md#what-other-env-files-can-i-use), so it is possible to have separate **.env**, **.env.local**, **.env.production**, etc. files and they will load according to the standard priority.
+
+You may choose to commit the default **.env** file or other standard configurations to version control, but generally **.env.local** files should be added to your **.gitignore**:
+
+```bash .gitignore
+.env*.local
+```
+
+If you create an environment-specific file, like **.env.test**, you can load it by setting `NODE_ENV` when running the Expo CLI:
 
-## Using Babel to inline environment variables during build time
+<Terminal cmd={['$ NODE_ENV=test npx expo start']} />
 
-An alternative approach is to replace all references of `process.env.VARNAME` with a value using a babel plugin. For example:
+## Environment variables in Expo Application Services
+### EAS Build
+[EAS Build](/build/introduction/) lets you define environment variables within build profiles in **eas.json**, as well as secrets for sensitive data required when building your native app. EAS Build can use **.env** files for inlining variables into the JavaScript embedded in your native app if they are made available to EAS Build, but you will still want to use **eas.json** and EAS Secrets for other environment-specific concerns, such as creating a separate test app variant or uploading source maps to an error-reporting service. Check out the EAS Build documentation on [environment variables and build secrets](/build-reference/variables.mdx) for more information.
```

Comment:
EAS CLI doesn't adhere to Expo's environment variable system, the only time there's overlap is when EAS CLI invokes Expo CLI -> Metro. The phrasing here could be more like the EAS Update section.